### PR TITLE
Deploying and linking libraries in Web3 context

### LINKED ISSUE #37: Contract needs to be linked to a library, this is only supported in the JavaScript VM for now.
Context: https://github.com/ethereum/browser-solidity/blob/master/src/universal-dapp.js#L397

Is there any reason why this can't or shouldn't be supported when in __Web3 Provider__ mode? I'm working on library support and could investigate this.

### PR DESCRIPTION
**Pull request by redsquirrel**
This is an attempt to resolve #37.

__Note:__ Even though it deploys and links the library, the bytecode in the UI does not update. (It still has the unlinked version.)

### CODE REVIEW

File: src/universal-dapp.js [hunk 1]
```diff
@@ -562,24 +556,7 @@ UniversalDApp.prototype.runTx = function( data, args, cb) {
         }
     }
 
-    if (!this.vm) {
```
Comment (axic) [inline src/universal-dapp.js]: Can you please have this part of the change as a separate commit? It doesn't seem to be related, but rather cosmetical.

Comment (redsquirrel) [inline src/universal-dapp.js]: @axic Yes, good call. I just forced pushed a simpler commit.

**Comment by axic** [comment]
I think #63 is the right way to handle `tryTillResponse()`. What do you think?

**Comment by redsquirrel** [comment]
Re: #63, that seems like a great update for `tryTillFinished`. The purpose of this PR is to remedy #37.

**Comment by axic** [comment]
Absolutely. You did change where `tryTrillFinished` is used, hence my comment. If it gets merged, you'll need to update this PR though.

**Comment by redsquirrel** [comment]
The squirrel with no name is ready.

**Comment by chriseth** [comment]
Did not work 100% for me, but probably still better than nothing. Also it would be good to get same feedback about how many libraries still need to be deployed.

**Comment by axic** [comment]
@chriseth @redsquirrel I think merging this is breaking after web3 was moved to the context. `tryTillResponse` references an invalid `this` now?

**Comment by redsquirrel** [comment]
@axic you're right, `this` should be `self`.
